src/main/java/xyz/nucleoid/creator_tools/command/MapMetadataCommand.java

```diff
@@ -64,13 +69,119 @@ public final class MapMetadataCommand {
+        static RegionSelector special(SpecialRegionSelector selector) {
+            return new RegionSelector(Either.right(selector));
+        }
+
+        static RegionSelector named(String name) {
+            return new RegionSelector(Either.left(name));
+        }
+
+        public boolean matches(WorkspaceRegion region) {
+            // This is to make sure we don't miss this if we add more special selectors:
+            // noinspection ConstantValue
+            return this.inner.map(
+                    name -> region.marker().equals(name),
+                    special -> switch (special) {
+                        case ALL -> true;
+                    }
+            );
+        }
+    }
+
+    /**
+     * Get a region marker argument, throwing an error if the name is reserved.
+     *
+     * @throws CommandSyntaxException if the name is reserved
+     */
+    private static String getRegionMarkerArg(CommandContext<ServerCommandSource> ctx, String argName) throws CommandSyntaxException {
+        var str = StringArgumentType.getString(ctx, argName);
+
+        if (str.startsWith(SpecialRegionSelector.PREFIX)) {
+            throw RESERVED_REGION_NAME.create(str);
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * Create a region marker argument. This is different to a region _selector_ argument, as it must be a valid name
+     * (e.g. as used in commit or rename). It still suggests currently existing region names for convenience.
+     */
+    private static RequiredArgumentBuilder<ServerCommandSource, String> regionMarkerArg(String argName) {
+        return argument(argName, StringArgumentType.word()).suggests(globalRegionSuggestions(false));
+    }
+
+    /**
+     * Create a region selector argument, which can either be a name or a special selector like +all.
+     */
+    @SuppressWarnings("SameParameterValue") // We want to keep this general
+    private static RegionSelector getRegionSelectorArg(CommandContext<ServerCommandSource> ctx, String name) throws CommandSyntaxException {
+        var str = StringArgumentType.getString(ctx, name);
+
+        if (str.startsWith(SpecialRegionSelector.PREFIX)) {
+            var selector = Arrays.stream(SpecialRegionSelector.values())
+                    .filter(s -> s.name.equals(str.substring(1)))
+                    .findAny()
+                    .orElseThrow(() -> INVALID_REGION_SELECTOR.create(SpecialRegionSelector.PREFIX + str));
+            return RegionSelector.special(selector);
+        } else {
+            return RegionSelector.named(str);
+        }
+    }
+
+    private static RequiredArgumentBuilder<ServerCommandSource, String> regionSelectorArg(
+            String name,
+            SuggestionProvider<ServerCommandSource> suggestions
+    ) {
+        return argument(name, StringArgumentType.word()).suggests(suggestions);
+    }
+
+    @SuppressWarnings("SameParameterValue") // We want to keep this general
```

Comment:
Nit: I believe putting this as a public helper in another class would disable that inspection - might generally be nice to have in separate classes as [we are doing in Plasmid](https://github.com/NucleoidMC/plasmid/tree/1.20.4/src/main/java/xyz/nucleoid/plasmid/command/argument)